### PR TITLE
Search: Fixing focus state for search inputs

### PR DESCRIPTION
**Pull request by shaunandrews**
Search inputs currently don't have a normal `:focus` state like other inputs in Calypso:

screen shot 2018-01-26 at 10 46 59 am

This PR aims to resolve that, ending up with this:

screen shot 2018-01-26 at 10 47 28 am

This will affect all search inputs. Themes requires a little more finesse, as it uses `sticky-panel`, which has its own quirks.

### CODE REVIEW

**Comment by folletto** [comment]
Tested on Chrome:

1. reader search — ✅ 
2. reader → manage — ✅ 
3. switch sites — ✅  (I thought it wouldn't look nice, but seems ok to me)
4. blog posts search — ✅ 
5. pages search — ✅ 
6. themes search — ✅ 
7. themes search, sticky when scrolling — ✅ 
8. me → billing history (2 forms) — ✅ 

Seems good to me!

File: client/components/sticky-panel/style.scss [hunk 1]
```diff
@@ -1,7 +1,10 @@
 .sticky-panel {
 	// Force a new block formatting context to prevent margin collapsing. It's needed for
 	// correct measurement of `.sticky_panel__content` height and setting the spacer height.
-	overflow: hidden;
+
+	.is-section-stats & {
```
Comment (folletto) [inline client/components/sticky-panel/style.scss]: I wonder if we should add a comment here on why it's specific to Stats?

**Comment by folletto** [comment]
Code seems good too, I just left a question. Feel free to push forward regardless.

**Comment by rachelmcr** [comment]
The themes search feels a bit off to me — the focus style stays on the search bar even when I tab to a different element that now actually has the focus. For example, in this screenshot I started typing in the search bar and then tabbed down to the "Upload Theme" button and both have focus styles:

screenshot 2018-01-26 16 26 46

**Comment by ryelle** [comment]
Thanks for fixing this! There's some history I dug up in #19283 – the accessible focus class _usually_ works, but for inputs it's a different story due to what keyboard events it listens for. 

I also noticed this in themes search – the focus on the X is cut off a little.

screen shot 2018-01-26 at 11 40 43 am

File: client/components/sticky-panel/style.scss (hunk 1 above)
Comment (shaunandrews) [reply client/components/sticky-panel/style.scss]: Heh, if I only knew. I'll add something.

**Comment by michaeldcain** [comment]
LGTM. Nice work!

**Comment by shaunandrews** [comment]
> The themes search feels a bit off to me — the focus style stays on the search bar even when I tab to a different element that now actually has the focus.

Been spinning my wheels a bit on this — it's because the `magic-search` component uses its own `.is-highlighted` class, which is currently toggled when you click inside/outside of the `magic-search` element. This means navigating outside of the `magic-search` element with the keyboard doesn't trigger that toggle.

@lsinger maybe you have an idea of how to fix this?

**Comment by folletto** [comment]
If @budzanowski has a moment might be able to give some insight.

FWIW I'd try to merge this asap and review that behaviour separately as it's a different issue that I guess it wasn't introduced here. :)

**Comment by mattsherman** [comment]
Nice! Glad to see this addressed. 🎉

**Pull request merged**